### PR TITLE
Legs

### LINKED ISSUE #11: Mimic Mapbox Directions API
In order to use all the functionality from Mapbox in the mobile apps, it would be best if we could mimic the response from [Mapbox directions API](https://www.mapbox.com/api-documentation/#directions).

A few issues still remain regarding the possible use of Mapbox navigation library (for mobiles) currently in Beta.

### PR DESCRIPTION
**Pull request by carocad**
- fixes #11 
- return the route-legs if requested by the user
- replace use of map-centered :lat :lon to generic protocol GeoCoordinate
- add a way to transform a fake graph into a network
- return the waypoint names on the response

### CODE REVIEW

**Comment by carocad** [comment]
bocked by #37

**Comment by carocad** [comment]
@mehdisadeghi this is ready for review. It is far from perfect but I think it is enough for the time being. Specially the turn-by-turn instruction are quite crappy.

**Comment by carocad** [comment]
BTW: I added some lines at the end of the `directions.clj` file that create a geojson file (in `resources/linestring.json`) based on the output of the direction function.

You can visualize the output here: http://geojsonlint.com/

pretty cool :)

**Comment by mehdisadeghi** [comment]
> BTW: I added some lines at the end of the directions.clj file that create a geojson file (in resources/linestring.json) based on the output of the direction function.

That was cool! I realized that you have used a fixed origin and destination, so no matter how many times I run the code, I'll get the same output, right?

**Comment by carocad** [comment]
@mehdisadeghi yeah I fixed them because I am still testing/improving the turn by turn navigation. There is also some code to randomly select a point. See the commented lines that start with `juxt`

**Comment by carocad** [comment]
@mehdisadeghi I just fixed the turn by turn instructions. They should be much more accurate rigth now :)

**Comment by carocad** [comment]
@mehdisadeghi by the way. This makes the generated graph useless since now we need a much bigger thing (a `network`) to represent the complete road*network*. 

It currently throws an error if you try to run it as it is.

File: src/service/routing/directions.clj [hunk 1]
```diff
@@ -1,10 +1,31 @@
...
             [service.routing.graph.protocols :as rp]
-            [service.routing.utils.math :as math]
-            [service.routing.spec :as specs]))
+            [service.routing.libs.math :as math]))
+            ;[service.routing.graph.generators :as g]))
+            ;[clojure.test.check.generators :as gen]))
+            ;[cheshire.core :as cheshire]))
+
+;; https://www.mapbox.com/api-documentation/#stepmaneuver-object
+(def bearing-turns
```
Comment (mehdisadeghi) [inline src/service/routing/directions.clj]: @carocad are these values fixed or shall we calculate them in future?

Comment (carocad) [reply src/service/routing/directions.clj]: no, I just estimated them. Pure empirism. If you find a way to calculate them or even a reference with a standard for those. I am more than happy to change them

**Pull request merged**